tests/test_encoding/test_decision_tree_encoder.py

```diff
@@ -163,3 +222,119 @@ def test_assigns_param_grid(grid):
         assert encoder._assign_param_grid() == {"max_depth": [1, 2, 3, 4]}
     else:
         assert encoder._assign_param_grid() == grid
+
+
+def test_unseen_is_encode(df_enc):
+    encoder = DecisionTreeEncoder(unseen="encode", regression=False, fill_value=-1)
+    encoder.fit(df_enc[["var_A", "var_B"]], df_enc["target"])
+
+    X_unseen_input = pd.DataFrame(
+        {
+            "var_A": ["A", "ZZZ", "YYY"],
+            "var_B": ["C", "YYY", "ZZZ"],
+        }
+    )
+
+    X_unseen_output = pd.DataFrame(
+        {
+            "var_A": [0.25, -1, -1],
+            "var_B": [0.4, -1, -1],
+        }
+    )
+
+    Xt = encoder.transform(X_unseen_input)
+    pd.testing.assert_frame_equal(Xt, X_unseen_output)
+
+
+def test_unseen_is_ignore(df_enc):
+    encoder = DecisionTreeEncoder(unseen="ignore", regression=False)
+    encoder.fit(df_enc[["var_A", "var_B"]], df_enc["target"])
+
+    X_unseen_input = pd.DataFrame(
+        {
+            "var_A": ["A", "ZZZ", "YYY"],
+            "var_B": ["C", "YYY", "ZZZ"],
+        }
+    )
+
+    X_unseen_output = pd.DataFrame(
+        {
+            "var_A": [0.25, np.nan, np.nan],
+            "var_B": [0.4, np.nan, np.nan],
+        }
+    )
+
+    Xt = encoder.transform(X_unseen_input)
+    pd.testing.assert_frame_equal(Xt, X_unseen_output)
+
+
+def test_fit_errors_if_new_cat_values_and_unseen_is_raise_param(df_enc):
+    encoder = DecisionTreeEncoder(unseen="raise", regression=False)
+    encoder.fit(df_enc[["var_A", "var_B"]], df_enc["target"])
+    X = pd.DataFrame(
+        {
+            "var_A": ["A", "ZZZ", "YYY"],
+            "var_B": ["C", "YYY", "ZZZ"],
+        }
+    )
+    var_ls = "var_A, var_B"
+    msg = (
+        "During the encoding, NaN values were introduced in the feature(s) "
+        f"{var_ls}."
+    )
+    # new categories will raise an error
+    with pytest.raises(ValueError) as record:
```

Comment:
Also look at this one

Reply:
Yep, I didn't miss them. I changed them to match=msg, but then these 2 returned an error which I didn't understand and I was lazy to resolve. Although the messages were identical, the test failed, and it said something about did you want to **escape** something? Did you come across this before? Probably something to do with regex. which is not my forte :/

Reply:
both lines of the msg should be f-strings in both cases

Reply:
Hmm, not sure I follow: This is the error

```
E       AssertionError: Regex pattern did not match.
E        Regex: 'During the encoding, NaN values were introduced in the feature(s) var_A, var_B.'
E        Input: 'During the encoding, NaN values were introduced in the feature(s) var_A, var_B.'
E        Did you mean to `re.escape()` the regex?
```

And this is the code:

```
def test_fit_errors_if_new_cat_values_and_unseen_is_raise_param(df_enc):
    encoder = DecisionTreeEncoder(unseen="raise", regression=False)
    encoder.fit(df_enc[["var_A", "var_B"]], df_enc["target"])
    X = pd.DataFrame(
        {
            "var_A": ["A", "ZZZ", "YYY"],
            "var_B": ["C", "YYY", "ZZZ"],
        }
    )
    var_ls = "var_A, var_B"
    msg = (
        f"During the encoding, NaN values were introduced in the feature(s) {var_ls}."
    )
    # new categories will raise an error
    with pytest.raises(ValueError, match=msg):
        encoder.transform(X)
    # assert str(record.value) == msg
```

What am I doing wrong?

Reply:
Oh, I see. Match is a regex patter search, so it gets confused by `(s)` put an escape characters like this

`f"During the encoding, NaN values were introduced in the feature\(s\) {var_ls}."`

Reply:
I think it is the parentesis "()".
I assume the match of the msg uses some regex, and having "()" within the string trigger some regex patterns and throws the exception.

Reply:
Updated the comment